java/impl/gcp/src/main/java/co/worklytics/psoxy/SecretManagerConfigService.java

```diff
@@ -71,6 +72,25 @@ public void putConfigProperty(ConfigProperty property, String value) {
                 SecretVersion version = client.addSecretVersion(secretName, payload);
 
                 log.info(String.format("Property: %s, stored version %s", secretName, version.getName()));
+
+                SecretManagerServiceClient.ListSecretVersionsPage enabledVersions = client.listSecretVersions(ListSecretVersionsRequest.newBuilder()
+                                .setFilter("state:ENABLED")
+                                .setParent(secretName.toString())
+                                // Reduce the page, as each version will be disabled one by one
+                                .setPageSize(20)
+                                .build())
+                        .getPage();
+
+                // Disable secret version, just the first page
+                enabledVersions.getValues().forEach(i -> {
+                    if (!i.getName().equals(version.getName())) {
+                        client.disableSecretVersion(DisableSecretVersionRequest.newBuilder()
+                                .setName(i.getName())
+                                .build());
```

Comment:
I'd wrap this in a try-catch or better yet extract a method `disableSecretVersions(SecretVersion versionToKeep)` with proper error handling just to make sure that a failure disabling won't affect the outcome of the method (specially if the new version was successfully added)